src/geometry.rs

```diff
@@ -1,9 +1,115 @@
 //! Extension's to Smithay's geometry module.
 
-use std::cmp;
+use std::error::Error;
+use std::str::FromStr;
+use std::{cmp, fmt, ops};
 
 use smithay::utils::{Point, Size};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Matrix3x3<T: Copy> {
+    storage: Vec<T>,
```

Comment:
Couldn't this just as well be a `[T; 9]`?

Reply:
I've decided to use something that consumes less stack given that it'll be static.